src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+        /// <returns>An object whose value is the sum of the date represented by this instance and the number of years represented by value.</returns>
+        public DateOnly AddYears(int value) => new DateOnly(DayNumberFromDateTime(GetEquivalentDateTime().AddYears(value)));
+
+        /// <summary>
+        /// Determines whether two specified instances of DateOnly are equal.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left and right represent the same date; otherwise, false.</returns>
+        public static bool operator ==(DateOnly left, DateOnly right) => left._dayNumber == right._dayNumber;
+
+        /// <summary>
+        /// Determines whether one specified DateOnly is later than another specified DateTime.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is later than right; otherwise, false.</returns>
+        public static bool operator >(DateOnly left, DateOnly right) => left._dayNumber > right._dayNumber;
+
+        /// <summary>
+        /// Determines whether one specified DateOnly represents a date that is the same as or later than another specified DateOnly.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is the same as or later than right; otherwise, false.</returns>
+        public static bool operator >=(DateOnly left, DateOnly right) => left._dayNumber >= right._dayNumber;
+
+        /// <summary>
+        /// Determines whether two specified instances of DateOnly are not equal.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left and right do not represent the same date; otherwise, false.</returns>
+        public static bool operator !=(DateOnly left, DateOnly right) => left._dayNumber != right._dayNumber;
+
+        /// <summary>
+        /// Determines whether one specified DateOnly is earlier than another specified DateOnly.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is earlier than right; otherwise, false.</returns>
+        public static bool operator <(DateOnly left, DateOnly right) => left._dayNumber < right._dayNumber;
+
+        /// <summary>
+        /// Determines whether one specified DateOnly represents a date that is the same as or earlier than another specified DateOnly.
```

Comment:
Should reference to types use hrefs? 

cc: @gewarren could you help reviewing the triple slash comments?

Reply:
@gewarren you are welcome to review the comments but I think this will be too much discuss in this PR.